test/functional/wallet_basic.py

```diff
@@ -252,9 +252,15 @@ def run_test(self):
         assert_raises_rpc_error(-6, "Fee rate (0.999 sat/vB) is lower than the minimum fee rate setting (1.000 sat/vB)",
             self.nodes[2].sendmany, amounts={address: 10}, fee_rate=0.99999999)
 
-        self.log.info("Test sendmany raises if fee_rate of 0 or -1 is passed")
-        assert_raises_rpc_error(-6, "Fee rate (0.000 sat/vB) is lower than the minimum fee rate setting (1.000 sat/vB)",
-            self.nodes[2].sendmany, amounts={address: 10}, fee_rate=0)
+        self.log.info("Test sendmany raises RPC error when invalid fee rates are passed")
+        msg = "Invalid amount"
+        # Test zero and values non-representable by CFeeRate
+        for invalid_value in ["0", 0, 0.00000001, 0.0009, 0.00099999]:
+            assert_raises_rpc_error(-3, msg, self.nodes[2].sendmany, amounts={address: 10}, fee_rate=invalid_value)
+        # Test values rejected by ParseFixedPoint() called in AmountFromValue()
+        for invalid_value in ["", 0.000000001, 1.111111111, 11111111111]:
+            assert_raises_rpc_error(-3, msg, self.nodes[2].sendmany, amounts={address: 1.0}, fee_rate=invalid_value)
```

Comment:
Am testing CFeeRate errors separately from AmountFromValue errors, as a follow-up PR will improve the messages of both.